tests/functional/event_handler/required_dependencies/appsync/test_appsync_batch_resolvers.py

```diff
@@ -942,3 +649,181 @@ async def get_locations2(event: AppSyncResolverEvent, name: str) -> str:
+            },
+        },
+        {
+            "typeName": "Query",
+            "info": {
+                "fieldName": "listLocations",
+                "parentTypeName": "Post",
+            },
+            "fieldName": "listLocations",
+            "arguments": {},
+            "source": {
+                "id": [3, 4],
+            },
+        },
+    ]
+
+    app = AppSyncResolver()
+
+    # WHEN the batch resolver for the listLocations field is defined
+    @app.async_batch_resolver(field_name="listLocations")
+    async def create_something(event: List[AppSyncResolverEvent]) -> Optional[list]:  # noqa AA03 VNE003
+        results = []
+        for record in event:
+            results.append(record.source.get("id") if record.source else None)
+
+        return results
+
+    # THEN the resolver should correctly process the batch of queries
+    result = app.resolve(event, LambdaContext())
+    assert result == [appsync_event["source"]["id"] for appsync_event in event]
+
+    assert app.current_batch_event and len(app.current_batch_event) == len(event)
+    assert not app.current_event
+
+
+def test_resolve_batch_processing_with_aggregate_and_returning_a_non_list():
+    # GIVEN a list of events representing GraphQL queries for listing locations
+    event = [
+        {
+            "typeName": "Query",
+            "info": {
+                "fieldName": "listLocations",
+                "parentTypeName": "Post",
+            },
+            "fieldName": "listLocations",
+            "arguments": {},
+            "source": {
+                "id": "1",
+            },
+        },
+    ]
+
+    app = AppSyncResolver()
+
+    # WHEN the batch resolver for the listLocations field is defined
+    @app.batch_resolver(field_name="listLocations")
+    def create_something(event: List[AppSyncResolverEvent]) -> Optional[list]:  # noqa AA03 VNE003
+        return event[0].source.get("id") if event[0].source else None
+
+    # THEN the resolver should raise a RuntimeError when processing the batch of queries
+    with pytest.raises(InvalidBatchResponse):
+        app.resolve(event, LambdaContext())
+
+
+def test_resolve_async_batch_processing_with_aggregate_and_returning_a_non_list():
+    # GIVEN a list of events representing GraphQL queries for listing locations
+    event = [
+        {
+            "typeName": "Query",
+            "info": {
+                "fieldName": "listLocations",
+                "parentTypeName": "Post",
+            },
+            "fieldName": "listLocations",
+            "arguments": {},
+            "source": {
+                "id": "1",
+            },
+        },
+    ]
+
+    app = AppSyncResolver()
+
+    # WHEN the batch resolver for the listLocations field is defined
+    @app.async_batch_resolver(field_name="listLocations")
+    async def create_something(event: List[AppSyncResolverEvent]) -> Optional[list]:  # noqa AA03 VNE003
+        return event[0].source.get("id") if event[0].source else None
```

Comment:
idea: have one test with where you don't return anything, simply process whatever. It's something our customers would end up accidentally having too.